src/routes/transactions/transactions-view.service.ts

```diff
@@ -61,12 +63,16 @@ export class TransactionsViewService {
         to: args.transactionDataDto.to,
       })
       .catch(() => {
-        // TODO: Get Kiln to verify all deployments
         // Fallback for unverified contracts
-        return {
-          method: '',
-          parameters: null,
-        };
+        const { data } = args.transactionDataDto;
+        const fallbackDataDecoded =
+          this.kilnDecoder.decodeDeposit(data) ??
+          this.kilnDecoder.decodeValidatorsExit(data) ??
+          this.kilnDecoder.decodeBatchWithdrawCLFee(data);
+        if (!fallbackDataDecoded) {
+          throw new Error('Transaction data could not be decoded');
+        }
+        return fallbackDataDecoded;
```

Comment:
Do you think we should assign the fallback value here, or in the mappings? The reason being, `deposit` is a very generic signature and could be returned for non-Kiln interactions.

I suggest we also add a test case for "generic" signatures as well. What do you think?

Reply:
I've moved the fallback decoding to the specific private function for each transaction type in [3fe322c](https://github.com/safe-global/safe-client-gateway/pull/1903/commits/3fe322ce083219a23b36488475b54d71191bc49b)